torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2721,25 +2720,25 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::collective(
   //
   // See [Sync Streams].
   if (!avoidRecordStreams) {
-    if (!input.is_sparse()) {
+    if (!inputs[0].is_sparse()) {
       c10::cuda::CUDACachingAllocator::recordStream(
-          input.storage().data_ptr(), ncclStream);
+          inputs[0].storage().data_ptr(), ncclStream);
     } else {
       // for sparse input case record streams on both index and value
       // tensors
       c10::cuda::CUDACachingAllocator::recordStream(
-          input.values().storage().data_ptr(), ncclStream);
+          inputs[0].values().storage().data_ptr(), ncclStream);
       c10::cuda::CUDACachingAllocator::recordStream(
-          input.indices().storage().data_ptr(), ncclStream);
+          inputs[0].indices().storage().data_ptr(), ncclStream);
     }
   }
 #ifndef NCCL_HAS_COMM_NONBLOCKING
   C10D_NCCL_CHECK(
-      fn(input, output, comm, ncclStream),
+      fn(inputs[0], outputs[0], comm, ncclStream),
       ncclComm->getNcclCommFailureReason());
 #else
   C10D_NCCL_CHECK_TIMEOUT(
-      fn(input, output, comm, ncclStream),
+      fn(inputs[0], outputs[0], comm, ncclStream),
       comm,
       ncclComm->getNcclCommFailureReason());
 #endif
```

Comment:
I think this style is still awkward. 

For record stream, we can probably iterate over every input to make it look good.

For fn(...), I don't know what style makes sense.. The definition of lambda itself has a mismatch with what input(s) are actually used -- the actual inputs are recorded rather than passed in..

Reply:
yep... I propose to keep the fn as it is for now.